pkg/controller/repository/repository_controller.go

```diff
@@ -237,140 +245,126 @@ func (r *ReconcileRepository) assertDeletedRepository(log logr.Logger, harborCli
+	projectMeta *registriesv1alpha1.RepositoryMetadata) *modelv1.ProjectMetadata {
+	pm := modelv1.ProjectMetadata{
+		AutoScan:             helper.BoolToString(projectMeta.AutoScan),
+		EnableContentTrust:   helper.BoolToString(projectMeta.EnableContentTrust),
+		PreventVul:           helper.BoolToString(projectMeta.PreventVul),
+		Public:               helper.BoolToString(projectMeta.Public),
+		ReuseSysCveWhitelist: helper.BoolToString(projectMeta.ReuseSysSVEWhitelist),
+		Severity:             projectMeta.Severity,
 	}
-	return pm, nil
+
+	return &pm
 }
 
-// reconcileProjectMembers reconciles the user-defined project members for a repository based on the actual Harbor user
-func (r *ReconcileRepository) reconcileProjectMembers(repository *registriesv1alpha1.Repository, harborClient *h.Client, heldRepository *h.Project) error {
-	members, err := harborClient.Projects().GetProjectMembers(heldRepository.ProjectID)
+// reconcileProjectMembers reconciles the user-defined project members for a repository based on an existing Harbor user
+func (r *ReconcileRepository) reconcileProjectMembers(ctx context.Context, repository *registriesv1alpha1.Repository,
+	harborClient *h.RESTClient, heldProject *modelv1.Project) error {
+	// List project members
+	members, err := harborClient.ListProjectMembers(ctx, heldProject)
 	if err != nil {
 		return err
 	}
 
+	// Range over the defined member users of a project
 	for _, memberRequestUser := range repository.Spec.MemberRequests {
+		// Fetch the user resource
 		user, err := r.getUserFromRef(memberRequestUser.User, repository.Namespace)
 		if err != nil {
 			return err
 		}
 
-		harborUser, err := internal.GetUser(user, harborClient)
+		// Check for an existing user
+		harborUser, err := harborClient.GetUser(ctx, user.Spec.Name)
 		if err != nil {
 			return err
 		}
 
 		roleID := memberRequestUser.Role.ID()
 
-		projectMember := h.MemberReq{
-			Role: roleID,
-			MemberUser: h.User{
-				Username: harborUser.Username,
-				UserID:   harborUser.UserID,
-			},
-		}
-
 		member := getMemberUserFromList(members, user)
 
 		if member == nil {
-			err = harborClient.Projects().AddProjectMember(heldRepository.ProjectID, projectMember)
+			err = harborClient.AddProjectMember(ctx, heldProject, harborUser, int(roleID))
 			if err != nil {
 				return err
 			}
-			break
-		}
 
-		// Update the project member
-		role, err := harborClient.Projects().GetProjectMember(heldRepository.ProjectID, int64(member.ID))
-		if err != nil {
-			return err
-		}
-
-		if roleID == role.RoleID {
 			break
-		}
-
-		err = harborClient.Projects().UpdateProjectMember(
-			heldRepository.ProjectID,
-			int64(member.ID),
-			h.RoleRequest{Role: roleID})
-		if err != nil {
-			return err
+		} else {
+			// Update the project member
+			if roleID == registriesv1alpha1.MemberRoleID(member.RoleID) {
+				break
+			}
+			err = harborClient.UpdateProjectMemberRole(ctx, heldProject, harborUser, int(roleID))
+			if err != nil {
```

Comment:
This assigns to an `err` variable in the parent scope. Maybe:

```suggestion
			if err := harborClient.UpdateProjectMemberRole(ctx, heldProject, harborUser, int(roleID)); err != nil {
```